odo/numpy_dtype.py

```diff
@@ -24,7 +24,8 @@ def unit_to_dtype(ds):
         ds = ds.measure
     if isinstance(ds, Option) and isscalar(ds) and isnumeric(ds):
         return unit_to_dtype(str(ds).replace('int', 'float').replace('?', ''))
-    if isinstance(ds, Option) and ds.ty in (date_, datetime_, string):
+    if isinstance(ds, Option) and isinstance(ds.ty, (type(date_),
+            type(datetime_), type(string))):
```

Comment:
TimeDeltas actually don't seem to fail, because they seem to trigger the `if isinstance(ds, Option) and isscalar(ds) and isnumeric(ds):` logic and have the Option stripped off there. In other words, TimeDelta is treated as numeric where DateTime/String are not. So I can't actually seem to write a failing test for TimeDelta, but I can write one anyway just in case isnumeric ever fails for it in the future.

I can incorporate that into the isinstance logic, but I don't think it'll ever actually hit it because of what I just mentioned.